cpp/mrc/include/mrc/edge/edge_holder.hpp

```diff
@@ -58,12 +59,37 @@ class EdgeHolder
 
         if (this->check_active_connection(false))
         {
-            LOG(FATAL) << "A node was destructed which still had dependent connections. Nodes must be kept alive while "
-                          "dependent connections are still active";
+            std::stringstream msg;
+            msg << "EdgeHolder(" << this << ") "
+                << "A node was destructed which still had dependent connections. Nodes must be kept alive while "
+                   "dependent connections are still active\n"
+                << this->connection_info();
+
+#if defined(NDEBUG)
+            LOG(ERROR) << msg.str();
+#else
+            LOG(FATAL) << msg.str();
+#endif
```

Comment:
`DFATAL` exists already, so we could just use that!

> The `DFATAL` severity logs a `FATAL` error in [debug mode](https://google.github.io/glog/0.7.1/logging/#debugging-support) (i.e., there is no `NDEBUG` macro defined), but avoids halting the program in production by automatically reducing the severity to `ERROR`.

```suggestion
            bool const is_connected = m_connected_edge && m_connected_edge->is_connected();
            LOG(DFATAL)
                << "EdgeHolder(" << this << ") "
                << "A node was destructed which still had dependent connections. Nodes must be kept alive while "
                   "dependent connections are still active\n"
                << "m_owned_edge=" << m_owned_edge.lock()
                << "\tm_owned_edge_lifetime=" << m_owned_edge_lifetime
                << "\tm_connected_edge=" << m_connected_edge
                << "\tis_connected=" << is_connected
                << "\tcheck_active_connection=" << this->check_active_connection(false);
```